shared/src/ledger/mod.rs

```diff
@@ -14,6 +14,521 @@ pub mod queries;
 pub mod storage;
 pub mod vp_host_fns;
 
+use std::path::PathBuf;
+use std::str::FromStr;
+
 pub use namada_core::ledger::{
     gas, parameters, replay_protection, storage_api, tx_env, vp_env,
 };
+use namada_core::types::dec::Dec;
+use namada_core::types::ethereum_events::EthAddress;
+use tokio::sync::{RwLock, RwLockReadGuard, RwLockWriteGuard};
+
+use crate::ibc::core::ics24_host::identifier::{ChannelId, PortId};
+use crate::proto::Tx;
+use crate::sdk::args::{self, InputAmount, SdkTypes};
+use crate::sdk::masp::{ShieldedContext, ShieldedUtils};
+use crate::sdk::signing::{self, SigningTxData};
+use crate::sdk::tx::{
+    self, ProcessTxResponse, TX_BOND_WASM, TX_BRIDGE_POOL_WASM,
+    TX_CHANGE_COMMISSION_WASM, TX_IBC_WASM, TX_INIT_PROPOSAL,
+    TX_INIT_VALIDATOR_WASM, TX_RESIGN_STEWARD, TX_REVEAL_PK, TX_TRANSFER_WASM,
+    TX_UNBOND_WASM, TX_UNJAIL_VALIDATOR_WASM, TX_UPDATE_ACCOUNT_WASM,
+    TX_UPDATE_STEWARD_COMMISSION, TX_VOTE_PROPOSAL, TX_WITHDRAW_WASM,
+    VP_USER_WASM,
+};
+use crate::sdk::wallet::{Wallet, WalletIo, WalletStorage};
+use crate::types::address::Address;
+use crate::types::io::Io;
+use crate::types::key::*;
+use crate::types::masp::{TransferSource, TransferTarget};
+use crate::types::token;
+use crate::types::token::NATIVE_MAX_DECIMAL_PLACES;
+use crate::types::transaction::GasLimit;
+
+#[async_trait::async_trait(?Send)]
+/// An interface for high-level interaction with the Namada SDK
+pub trait Namada<'a>: Sized {
```

Comment:
nit: I'm not entirely happy with the name because of how it appears in the apps crate. It's amost always assigned to something called `context`.  I feel like it should be `NamadaCtx` or `NamadaApi` or `NamadaUtils` or something.

Reply:
The name `context` is a bit of a hangover from how we were naming instances of `Context` - this too is changeable. This trait is intended to be the primary interface for SDK users (who won't be aware of its `apps` crate usage and our naming distinctions), so I was thinking more about the best look/outcome in this context: https://github.com/anoma/namada-sdk-starter/blob/murisi/updating-sdk-usage/src/main.rs. Also, SDK users (and we) will probably be typing `&impl Namada<'a>` a lot, so I'm keen to minimize the name's length. That being said, I'm not too attached to any particular name, and we can always export/import `as`.